source/configure-components/index.html.md.erb

```diff
@@ -13,6 +13,10 @@ You can configure a component by:
 - [passing JavaScript configuration](#passing-javascript-configuration)
 - [adding HTML data attributes](#adding-html-data-attributes)
 
+Although [JavaScript configuration objects](#passing-javascript-configuration) are optional, configuration can still be provided or overridden by [HTML data attributes](#adding-html-data-attributes).
```

Comment:
The 'although' here doesn't really make sense any more now that we've moved this. I also think we should talk about Nunjucks options, even if they are using data attributes under the hood. I also think it's expected / implied that they work even if you don't pass JS config, so not sure if we need that?

What about something like:

```suggestion
Configuration passed via data attributes in the HTML or via Nunjucks macro options will take precedence over any JavaScript configuration.
```